src/crimson/osd/pg.h

```diff
@@ -274,7 +274,9 @@ class PG : public boost::intrusive_ref_counter<
     shard_services.remove_want_pg_temp(pgid.pgid);
   }
   void publish_stats_to_osd() final {
-    // Not needed yet
+    // crimson always run PG::get_stats when trying to publish stats,
+    // we call get_stats here just for making sure pg_info stats are updated.
+    get_stats();
```

Comment:
i'd suggest call `recovery_state.prepare_stats_for_publish()` directly instead using `get_stats()` for its side effects. also, shall we just return if this pg is not primary?